torchtune/data/_messages.py

```diff
@@ -709,32 +782,102 @@ def __init__(
         }
 
     def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
-        column_map = self.column_map or {}
-        key_input = column_map.get("input", "input")
-        key_instruction = column_map.get("instruction", "instruction")
-        key_output = column_map.get("output", "output")
-
+        key_input = self._column_map.get("input", "input")
         if key_input in sample and sample[key_input]:
             prompt = self.template["prompt_input"].format(
-                instruction=sample[key_instruction], input=sample[key_input]
+                instruction=sample[self._column_map["instruction"]],
+                input=sample[key_input],
             )
         else:
             prompt = self.template["prompt_no_input"].format(
-                instruction=sample[key_instruction]
+                instruction=sample[self._column_map["instruction"]]
             )
 
         messages = [
             Message(
                 role="user",
                 content=prompt,
-                masked=not self.train_on_input,
                 eot=True,
             ),
             Message(
                 role="assistant",
-                content=sample[key_output],
-                masked=False,
+                content=sample[self._column_map["output"]],
                 eot=True,
             ),
         ]
+        mask_messages(messages, self.masking_strategy)
         return {"messages": messages}
+
+
+def validate_messages(
```

Comment:
Similar q here: don't we have an existing `validate_messages` function?

Reply:
it was just moved down here I believe

Reply:
I moved `validate_messages` down to keep all the message transforms at one place.